### DISCUSSION TITLE
`Mutate` is resolving to `never` without `zustand/middleware/immer` imported

### DISCUSSION THREAD

**Discussion by ruvkr** [pmndrs/zustand#1649] [Q&A]
`Mutate` is resolving to `never` without `zustand/middleware/immer` imported.

![Screenshot_20230222_125153](https://user-images.githubusercontent.com/34903858/220552189-c23749da-0b40-479e-b164-caf230f7a710.png)
![Screenshot_20230222_125221](https://user-images.githubusercontent.com/34903858/220552207-dcb87141-03e3-400d-bfc5-9f63ea5d38e7.png)

https://codesandbox.io/s/hungry-jones-sl7f9r?file=/src/App.tsx:94-118

**Comment by dai-shi** [comment]
I think it's intentional; without immer middleware, it can't be typed correctly.

What's your use case of the typing? We generally don't need to type things explicitly.

**Reply by ruvkr** [reply]
I'm building some internal libs, where I have multiple seperate generic actions, I'm using the store api type there. Stores will be created later and will be provided by context

```ts
export const performAction = <T extends { slice: Data }>(
  api: StoreWithImmer<T>,
  ...
): void => {
  const { ... } = api.getState();
  api.setState(state => {
    ...
  });
};
```

**Reply by dai-shi** [reply]
> I'm building some internal libs

That sounds tough. You may need to lie some types. Not sure.
If I were to do something, I'd avoid depending on zustand/middleware/immer.
Middleware typings are generally tricky.
